backend/src/packages/business/business.service.ts

```diff
@@ -0,0 +1,116 @@
+import { type IService } from '~/libs/interfaces/interfaces.js';
+import { HttpCode, HttpError, HttpMessage } from '~/libs/packages/http/http.js';
+import { type OperationResult } from '~/libs/types/types.js';
+import { type BusinessRepository } from '~/packages/business/business.repository.js';
+
+import { UserGroupIds } from '../users/libs/enums/enums.js';
+import { BusinessEntity } from './business.entity.js';
+import {
+  type BusinessAddResponseDto,
+  type BusinessCreatePayload,
+  type BusinessEntity as BusinessEntityT,
+  type BusinessUpdateResponseDto,
+} from './libs/types/types.js';
+
+class BusinessService implements IService {
+  private businessRepository: BusinessRepository;
+
+  public constructor(businessRepository: BusinessRepository) {
+    this.businessRepository = businessRepository;
+  }
+
+  public async find(
+    id: number,
+  ): Promise<OperationResult<BusinessEntityT | null>> {
+    const business = await this.businessRepository.find(id);
+
+    return { result: business ? business.toObject() : null };
+  }
+
+  public async create({
+    payload,
+    owner,
+  }: BusinessCreatePayload): Promise<BusinessAddResponseDto> {
+    if (owner.groupId !== UserGroupIds.BUSINESS) {
```

Comment:
```suggestion
    if (owner.group.key !== UserGroupKey.BUSINESS) {
```
Storing database Ids in code isn't reliable. Applications could be deployed to different environments and each can have a different set of Id's. Keys - is what is supposed to be static.